src/components/data-management/SamplesTable.jsx

```diff
@@ -0,0 +1,120 @@
+import UploadStatus from 'utils/upload/UploadStatus';
+
+const SamplesTable = (props) => {
+  const { tableColumns, activeProjectUuid, height } = props;
+  const dispatch = useDispatch();
+  const [tableData, setTableData] = useState([]);
+  const projects = useSelector((state) => state.projects);
+  const samples = useSelector((state) => state.samples);
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]) || false;
+
+  useEffect(() => {
+    if (!activeProject || !samples[activeProject.samples[0]]) {
+      setTableData([]);
+      return;
+    }
+
+    // Set table data
+
+    const newData = activeProject.samples.map((sampleUuid, idx) => {
+      const sampleFiles = samples[sampleUuid].files;
+
+      const barcodesFile = sampleFiles['barcodes.tsv.gz'] ?? { upload: { status: UploadStatus.FILE_NOT_FOUND } };
+      const genesFile = sampleFiles['features.tsv.gz'] ?? { upload: { status: UploadStatus.FILE_NOT_FOUND } };
+      const matrixFile = sampleFiles['matrix.mtx.gz'] ?? { upload: { status: UploadStatus.FILE_NOT_FOUND } };
+
+      const barcodesData = { sampleUuid, file: barcodesFile };
+      const genesData = { sampleUuid, file: genesFile };
+      const matrixData = { sampleUuid, file: matrixFile };
+
+      return {
+        key: idx,
+        name: samples[sampleUuid].name,
+        uuid: sampleUuid,
+        barcodes: barcodesData,
+        genes: genesData,
+        matrix: matrixData,
+        species: samples[sampleUuid].species || DEFAULT_NA,
+        ...samples[sampleUuid].metadata,
+      };
+    });
+    setTableData(newData);
+  }, [projects, samples, activeProjectUuid]);
+
+  const onSortEnd = ({ oldIndex, newIndex }) => {
+    if (oldIndex !== newIndex) {
+      // This can be done because there is only one experiment per project
+      // Has to be changed when we support multiple experiments per project
+      const experimentId = activeProject.experiments[0];
+
+      const newData = arrayMove([].concat(tableData), oldIndex, newIndex).filter((el) => !!el);
+      const newSampleOrder = newData.map((sample) => sample.uuid);
+
+      dispatch(updateProject(activeProjectUuid, { samples: newSampleOrder }));
+      dispatch(updateExperiment(experimentId, { sampleIds: newSampleOrder }));
+      setTableData(newData);
+    }
+  };
+
+  const SortableRow = sortableElement((otherProps) => <tr {...otherProps} className={`${otherProps.className} drag-visible`} />);
+  const SortableTable = sortableContainer((otherProps) => <tbody {...otherProps} />);
+
+  const DragContainer = (otherProps) => (
+    <SortableTable
+      useDragHandle
+      disableAutoscroll
+      helperClass='row-dragging'
+      onSortEnd={onSortEnd}
+      {...otherProps}
+    />
+  );
+
+  const DraggableRow = (otherProps) => {
+    const index = tableData.findIndex((x) => x.key === otherProps['data-row-key']);
+    return <SortableRow index={index} {...otherProps} />;
+  };
```

Comment:
can you move these components to not be nested inside `SamplesTable`, but rather be defined on the same level as `SamplesTable`? Better to put them on top

Reply:
it is quite confusing to reason about them and quickly realise which one depends on which and so on